packages/calypso-e2e/src/lib/pages/marketing-page.ts

```diff
@@ -56,32 +54,65 @@ export class MarketingPage {
 	/**
 	 * Enters text into the Website Meta Information field.
 	 *
-	 * @param {string} [text] String to be used as the description of the web site in SEO.
-	 * @returns {Promise<void>} No return value.
+	 * @param {string} text String to be used as the description of the web site in SEO.
 	 */
-	async enterWebsiteMetaInformation( text = 'test text' ): Promise< void > {
-		await this.page.fill( selectors.websiteMetaTextArea, text );
+	async enterExternalPreviewText( text: string ) {
+		await this.page.getByRole( 'textbox', { name: 'Front Page Meta Description' } ).fill( text );
 	}
 
 	/**
-	 * Open the preview of SEO changes.
+	 * Validates the external preview for the specified service contains the text.
 	 *
-	 * @returns {Promise<void>} No return value.
+	 * @param {SEOExternalServices} service External service.
+	 * @param {string} text Text to validate.
 	 */
-	async openSEOPreview(): Promise< void > {
-		const locator = this.page.locator( selectors.seoPreviewButton );
-		await locator.click();
-		await this.page.waitForSelector( selectors.seoPreviewPane );
+	async validateExternalPreview( service: SEOExternalServices, text: string ) {
+		await this.page.locator( '.vertical-menu__social-item' ).filter( { hasText: service } ).click();
+
+		await this.page.locator( '.seo-preview-pane__preview' ).getByText( text ).waitFor();
+	}
+
+	/**
+	 * Given an accessible name of the button, click on the button.
+	 *
+	 * @param {string} name Accessible name of the button.
+	 */
+	async clickButton( name: string ) {
+		await this.page.getByRole( 'button', { name: name } ).click();
+	}
+
+	/**
+	 * Enters the specified text into the input of the specified category, changing the
+	 * page title structure.
+	 *
+	 * @param {SEOPageTitleStructureCategories} category Category to modify.
+	 * @param {string} text Text to enter.
+	 */
+	async enterPageTitleStructure( category: SEOPageTitleStructureCategories, text: string ) {
+		const target = this.page
+			.locator( selectors.pageTitleStructureInput )
+			.filter( { has: this.page.getByText( category ) } )
+			.getByRole( 'textbox' );
+
+		await target.scrollIntoViewIfNeeded();
+
+		await target.fill( text );
 	}
 
 	/**
-	 * Close the preview of SEO changes.
+	 * Returns the preview text for the page title structure category.
 	 *
-	 * @returns {Promise<void>} No return value.
+	 * @param {SEOPageTitleStructureCategories} category Category to return preview text for.
+	 * @returns {Promise<string>} Preview text.
 	 */
-	async closeSEOPreview(): Promise< void > {
-		await this.page.click( selectors.seoPreviewPaneCloseButton );
-		await this.page.waitForSelector( selectors.seoPreviewButton );
+	async getPreviewTextForPageStructureCategory(
```

Comment:
**Suggestion (non-blocking):** Definitely not required, but what do you think of designing this function as a `validate` function? 

I was thinking more of the case where you already have a title, but then you change it. The element will already exist in the DOM, and at that point it's effectively a race against the React re-render. 😆  I think in the vast majority of cases (and we don't even test this case right now), the re-render will be fast enough, but using `waitFor` for a specific piece of text might be even more stable for future extensions. Up to you though! 😄

Reply:
Done, changed to a validate function instead.